src/logics/ArithLogic.cc

```diff
@@ -0,0 +1,1161 @@
+{
+    checkSortReal(args);
+    if (args.size() != 2) {
+        throw OsmtApiException("Division operation requires exactly 2 arguments");
+    }
+    if (isZero(args[1])) {
+        throw ArithDivisionByZeroException();
+    }
+    if (not isConstant(args[1])) {
+        throw LANonLinearException("Only division by constant is permitted in linear arithmetic!");
+    }
+    SimplifyConstDiv simp(*this);
+    vec<PTRef> args_new;
+    SymRef s_new;
+    simp.simplify(get_sym_Real_DIV(), args, s_new, args_new);
+    if (isRealDiv(s_new)) {
+        assert((isNumTerm(args_new[0]) || isPlus(args_new[0])) && isConstant(args_new[1]));
+        args_new[1] = mkRealConst(FastRational_inverse(getNumConst(args_new[1]))); //mkConst(1/getRealConst(args_new[1]));
+        return mkTimes(args_new);
+    }
+    PTRef tr = mkFun(s_new, std::move(args_new));
+    return tr;
+}
+
+PTRef ArithLogic::insertTerm(SymRef sym, vec<PTRef>&& terms)
+{
+    if (isNeg(sym))
+        return mkNeg(terms[0]);
+    if (isMinus(sym))
+        return mkMinus(std::move(terms));
+    if (isPlus(sym))
+        return mkPlus(std::move(terms));
+    if (isTimes(sym))
+        return mkTimes(std::move(terms));
+    if (isRealDiv(sym))
+        return mkRealDiv(std::move(terms));
+    if (isLeq(sym))
+        return mkLeq(terms);
+    if (isLt(sym))
+        return mkLt(terms);
+    if (isGeq(sym))
+        return mkGeq(terms);
+    if (isGt(sym))
+        return mkGt(terms);
+    if (extendedSignatureEnabled()) {
+        if (isMod(sym))
+            return mkMod(terms[0], terms[1]);
+        if (isIntDiv(sym))
+            return mkIntDiv(std::move(terms));
+    }
+    return Logic::insertTerm(sym, std::move(terms));
+}
+
+PTRef ArithLogic::mkConst(SRef s, const char* name)
+{
+    assert(strlen(name) != 0);
+    PTRef ptr = PTRef_Undef;
+    if (s == sort_REAL or s == sort_INT) {
+        char* rat;
+        if (s == sort_REAL)
+            opensmt::stringToRational(rat, name);
+        else {
+            if (not opensmt::isIntString(name))
+                throw OsmtApiException("Not parseable as an integer");
+            rat = strdup(name);
+        }
+        ptr = mkVar(s, rat);
+        // Store the value of the number as a real
+        SymId id = sym_store[getPterm(ptr).symb()].getId();
+        for (auto i = numbers.size(); i <= id; i++)
+            numbers.emplace_back(nullptr);
+        if (numbers[id] != nullptr) { delete numbers[id]; }
+        numbers[id] = new FastRational(rat);
+        free(rat);
+        markConstant(id);
+    } else
+        ptr = Logic::mkConst(s, name);
+    return ptr;
+}
+
+/**
+ * SMT-LIB style resolution of numeric constants.
+ * @param name string representation of the constant
+ * @return a PTRef of the correct sort representing the constant.
+ */
+PTRef ArithLogic::mkConst(char const * name) {
+    bool isIntegralForm = opensmt::isIntString(name);
+    bool isRealForm = opensmt::isRealString(name);
```

Comment:
Is `IntString` a subset of `RealString` or they are not compatible?

Reply:
A subset.  So for all strings s `isIntString(s)` -> `isRealString(s)`.